downstairs/src/region.rs

```diff
@@ -251,23 +252,29 @@ impl Inner {
         Ok(())
     }
 
-    /*
-     * Get rid of all block context rows except those that match the on-disk
-     * hash that is computed after a flush.
-     */
+    /// Get rid of all block context rows except those that match the on-disk
+    /// hash that is computed after a flush. For best performance, make sure
+    /// `extent_block_indexes_and_hashes` is sorted by block number before
+    /// calling this function.
     fn truncate_encryption_contexts_and_hashes(
         &mut self,
         extent_block_indexes_and_hashes: Vec<(usize, u64)>,
     ) -> Result<()> {
         let tx = self.metadb.transaction()?;
 
         let stmt = "DELETE FROM block_context where block == ?1 and on_disk_hash != ?2";
+        let mut stmt = tx.prepare_cached(stmt)?;
+
+        let n_blocks = extent_block_indexes_and_hashes.len();
 
+        cdt::extent__context__truncate__start!(|| n_blocks as u64);
```

Comment:
it might make sense to include the commit. I was mainly just using this as debug when figuring out that sorting made a big difference and I left it in.